### PR TITLE
Remove all Silicon deprecations

### PR DESCRIPTION
**Pull request by waynew**
### What does this PR do?

Removes the bits that  were deprecated for Silicon

### What issues does this PR fix or reference?

#60893 

### Previous Behavior

Warning messages would appear about deprecated features

### New Behavior

Deprecated features are gone and you get errors instead.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by waynew** [comment]
@cbosdo I *think* this is the correct way to adjust the tests for this deprecated removal - if there's a better way, please let me know!

**Review by cbosdo — DISMISSED** [review]
LGTM

File: changelog/60893.removed [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed deprecated virt.migrate_non_shared ,virt.migrate_non_shared_inc, ssh from virt.migrate, and python2/python3 args from salt.utils.thin.gen_min and .gen_thin
```
Comment (s0undt3ch) [inline changelog/60893.removed:1]:
```suggestion
Removed deprecated virt.migrate_non_shared, virt.migrate_non_shared_inc, ssh from virt.migrate, and python2/python3 args from salt.utils.thin.gen_min and .gen_thin
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: changelog/60893.removed (hunk 1 above)
Comment (waynew) [reply changelog/60893.removed]: Nice catch! :+1:

**Comment by cbosdo** [comment]
@rst0git could you have a look at this PR? you know that code better than anyone else :grin:

File: salt/modules/virt.py [hunk 2]
```diff
@@ -5477,145 +5477,12 @@ def define_vol_xml_path(path, pool=None, **kwargs):
...
-def migrate(vm_, target, ssh=False, **kwargs):
+def migrate(vm_, target, **kwargs):
     """
     Shared storage migration
 
     :param vm_: domain name
     :param target: target libvirt URI or host name
-    :param ssh: True to connect over ssh
 
        .. deprecated:: 3002
```
Comment (rst0git) [inline salt/modules/virt.py:5487]: Perhaps we could drop this `.. deprecated:: 3002` comment as well?

**Review by rst0git — APPROVED** [review]

**Pull request merged**